### ISSUE TITLE
Cannot import `generate` in my Sveltekit project

### ISSUE THREAD

**Issue by ywiyogo** [pdfme/pdfme#257]
### Describe the bug

I try to integrate pdfme into my Sveltekit project.
I copy paste the example code from the doc to the page.svelte and get an error `SyntaxError: Cannot use import statement outside a module`.

### To Reproduce

Please check the integration in this online Sveltekit demo: https://www.sveltelab.dev/sqnh3ssj6s6te4m, or copy the below code in page.svelte.

```
<script lang="ts">
  import { generate, BLANK_PDF } from '@pdfme/generator'

  const template = {
    basePdf: BLANK_PDF,
    schemas: [
      {
        a: {
          type: 'text',
          position: { x: 0, y: 0 },
          width: 10,
          height: 10
        },
        b: {
          type: 'text',
          position: { x: 10, y: 10 },
          width: 10,
          height: 10
        },
        c: {
          type: 'text',
          position: { x: 20, y: 20 },
          width: 10,
          height: 10
        }
      }
    ]
  }

  function genPdf() {
    let date_now = new Date()
    // Generate the PDF ticket
    const inputs = [{ a: 'a1', b: 'b1', c: 'c1' }];

    generate({ template, inputs }).then((pdf) => {
      console.log(pdf)
      // Browser
      const blob = new Blob([pdf.buffer], { type: 'application/pdf' })
      window.open(URL.createObjectURL(blob))
    })

    console.log('finish pdf')
  }
</script>

    <button  on:click={() => { genPdf() }}>Generate PDF</button>
```

### Expected behavior

No error `Cannot use import statement outside a module`

### Your Environment

```markdown
- pdfme package(@pdfme/generator or @pdfme/ui):2.2.0
- pdfme version:2.2.0
- Operating system: Linux
- Node.js version or Browser name & version: node v19.1.0, any browser
- @sveltejs/kit: 1.11.0
```

### Your Error Log

```shell
[vite] Error when evaluating SSR module /src/routes/(pages)/checkout/+page.svelte: failed to import "@pdfme/generator"
|- /node_modules/.pnpm/@pdfme+generator@2.2.0_@pdfme+common@2.2.0/node_modules/@pdfme/generator/dist/esm/src/index.js:1
import generate from './generate.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:74:18)
    at wrapSafe (node:internal/modules/cjs/loader:1128:20)
    at Module._compile (node:internal/modules/cjs/loader:1169:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1259:10)
    at Module.load (node:internal/modules/cjs/loader:1068:32)
    at Module._load (node:internal/modules/cjs/loader:909:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Internal server error: Cannot use import statement outside a module
```
```


### Additional context

_No response_

<!-- POLAR PLEDGE BADGE START -->
## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/pdfme) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/pdfme/pdfme/issues/257">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/pdfme/pdfme/issues/257/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/pdfme/pdfme/issues/257/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

**Comment by liviaalmeida** [comment]
Also experiencing the same issue when moving our codebase from CJS to ESM - I see it tries to import `{ generate }` from the right dist folder - `esm`, but for some reason it doesn't allow us to use it in ESM with a import statement

**Comment by IvanKztro** [comment]
same issue, 
"import type { Designer } from '@pdfme/ui';" is working.
but "import { generate, BLANK_PDF } from '@pdfme/generator';" is not working

**Comment by dsun-cp** [comment]
We were about to adopt this lib for real, and just ran into similar issue


```
(node:3622) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3622) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
xxxxxxxxx/node_modules/@pdfme/generator/dist/esm/src/index.js:1
import generate from './generate.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.16.1
```

For reference:

The code usage like this
```
import * as fs from 'fs';
import { Template, generate } from '@pdfme/generator';
import path from 'path';

const template: Template = JSON.parse(
  fs.readFileSync(path.join(__dirname, 'template.json'), 'utf8')
);

const inputs = [{
  accountNumber: 'Account number: 12345678',
}];

generate({ template, inputs }).then((pdf) => {
  fs.writeFileSync(path.join(__dirname, 'abc.pdf'), pdf);
});

```

the tsconfig.json is
```
{
  "compilerOptions": {
    "target": "ES2020",
    "module": "ES2020",
    "moduleResolution": "node",
    "types": [
      "node",
      "vite/client"
    ],
    "declaration": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,

    "strict": true,
    "skipLibCheck": true
  },
  "ts-node": {
    "esm": true,
    "experimentalSpecifierResolution": "node",
    "files": true
  }
}

```

**Comment by dsun-cp** [comment]
@hand-dot 

could this be the fix?

```
---
diff --git a/packages/generator/src/index.ts b/packages/generator/src/index.ts
index 71da4c2..423d112 100644
--- a/packages/generator/src/index.ts
+++ b/packages/generator/src/index.ts
@@ -1,4 +1,4 @@
-import generate from './generate.js';
+import generate from './generate';
 import {
   BLANK_PDF,
   DEFAULT_FONT_VALUE,
--
```

**Comment by hand-dot** [comment]
Hey @dsun-cp ,

> could this be the fix?

Sorry, I don't know... Can you check by yourself?

**Comment by zaro** [comment]
I am having similar problems with Remix. 

```
 💿  remix dev

 info  building...
 info  built (11s)
(node:137991) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node-18 --trace-warnings ...` to show where the warning was created)
/home/zaro/kenya/lawyer-docs/node_modules/@pdfme/common/dist/esm/src/index.js:1
import { DEFAULT_FONT_NAME, DEFAULT_FONT_SIZE, DEFAULT_ALIGNMENT, VERTICAL_ALIGN_TOP, VERTICAL_ALIGN_MIDDLE, VERTICAL_ALIGN_BOTTOM, DEFAULT_VERTICAL_ALIGNMENT, DEFAULT_LINE_HEIGHT, DEFAULT_CHARACTER_SPACING, DEFAULT_FONT_COLOR, DYNAMIC_FIT_VERTICAL, DYNAMIC_FIT_HORIZONTAL, DEFAULT_DYNAMIC_FIT, FONT_SIZE_ADJUSTMENT, MM_TO_PT_RATIO, PT_TO_MM_RATIO, PT_TO_PX_RATIO, BLANK_PDF, DEFAULT_FONT_VALUE, } from './constants.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
....
```

I am not sure what is happening, but from the error message it seems node thinks that the the @pdfme/common is not a ESM module.

@dsun-cp  I tried manually editing the files as you suggested and it doesn't make any differentce.

**Comment by zaro** [comment]
It seems the core of the problem is described very well here :

https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html


Because If I manually add  a  `package.json` like that: 
```json
{
  "type": "module"
}
```
in `node_modules/@pdfme/common/dist/esm/` , the error changes to : 

```
Error: Cannot find module '/home/zaro/kenya/lawyer-docs/node_modules/@pdfme/common/dist/esm/src/schema' imported from /home/zaro/kenya/lawyer-docs/node_modules/@pdfme/common/dist/esm/src/helper.js
```

Which is expected, since I think with ESM modules extension is necessery.

**Comment by hand-dot** [comment]
I would be happy to review it if you create a PR.

**Comment by UsmanQureshi** [comment]
(node:11446) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
../../node_modules/@pdfme/generator/dist/esm/src/index.js:1
import generate from './generate.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module

I do have "type":"module" in the project package.json. but seems the package.json for @pdfme generator is missing something..

here is dependency for
 "dependencies": {
    "@pdfme/generator": "^1.0.17",
    "bcrypt": "^3.0.6",
    "bcryptjs": "^2.4.3",
    "body-parser": "^1.19.0",
    "cors": "^2.8.4",
    "dotenv": "^8.6.0",
..

Also note this was working on different Linux machine when  i installed it around 6 months ago.

**Comment by AugustoResende** [comment]
I use typescript and my tsconfig is like that

```
{
  "compilerOptions": {
    /* Visit https://aka.ms/tsconfig to read more about this file */
    "target": "ES2020",
    "lib": ["ES2020"],
    "module": "ES2020",
    "outDir": "./dist",
    "rootDir": "./src",
    "removeComments": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "node",
    "strict": true,
    "skipLibCheck": true,
  },=
}

And to successfully use the lib i needed to apply this patch, to both common and generator

```
diff --git a/node_modules/@pdfme/common/package.json b/node_modules/@pdfme/common/package.json
index 2725927..df0fc38 100644
--- a/node_modules/@pdfme/common/package.json
+++ b/node_modules/@pdfme/common/package.json
@@ -22,14 +22,8 @@
     "url": "https://github.com/pdfme/pdfme/issues"
   },
   "main": "dist/cjs/src/index.js",
-  "module": "dist/esm/src/index.js",
+  "module": "dist/cjs/src/index.js",
   "types": "dist/types/src/index.d.ts",
-  "exports": {
-    ".": {
-      "import": "./dist/esm/src/index.js",
-      "require": "./dist/cjs/src/index.js"
-    }
-  },
   "engines": {
     "node": ">=16"
   },
```
```

**Comment by hand-dot** [comment]
You can pledge behind and help support this effort using Polar.sh


<a href="https://polar.sh/pdfme/pdfme/issues/257">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/pdfme/pdfme/issues/257/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/pdfme/pdfme/issues/257/pledge.svg">
</picture>
</a>

**Comment by elch-yan** [comment]
Hey, everyone, first of all thank you for the terrific library.

I also encountered the same esm/cjs related issue when trying to run my ES6 transpiled code, on AWS Lambda with node18 runtime.

When I've been trying to do ES6 lib import:
`import { generate } = '@pdfme/generator'`
 compiler was complaining about the module being commonjs, but when doing commonjs lib import for ES6:
```
import pkg from '@pdfme/generator';
const { generate } = pkg;
```
then, I was getting error about imports being used outside of the module detected in library's esm files.

I resolved the issue in a way almost identical to the one @AugustoResende suggested, only difference is that removing only following part from generators package.json was sufficient:
```
  "exports": {
    ".": {
      "import": "./dist/esm/src/index.js",
      "require": "./dist/cjs/src/index.js"
    }
  },
```
I'm not sure about the functionality aspect of that declaration, can removing it break anything? If not is it possible to have it removed? 

P.S. I would be happy to open a pull request if needed.

**Comment by Chianche** [comment]
I'd also be interested in a fix, I really like this library but cannot use it in my SvelteKit project❤️

**Comment by alejdig** [comment]
> > I use typescript and my tsconfig is like that
> > ```
> > {
> >   "compilerOptions": {
> >     /* Visit https://aka.ms/tsconfig to read more about this file */
> >     "target": "ES2020",
> >     "lib": ["ES2020"],
> >     "module": "ES2020",
> >     "outDir": "./dist",
> >     "rootDir": "./src",
> >     "removeComments": true,
> >     "esModuleInterop": true,
> >     "forceConsistentCasingInFileNames": true,
> >     "moduleResolution": "node",
> >     "strict": true,
> >     "skipLibCheck": true,
> >   },
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And to successfully use the lib i needed to apply this patch, to both common and generator
> > ```
> > diff --git a/node_modules/@pdfme/common/package.json b/node_modules/@pdfme/common/package.json
> > index 2725927..df0fc38 100644
> > --- a/node_modules/@pdfme/common/package.json
> > +++ b/node_modules/@pdfme/common/package.json
> > @@ -22,14 +22,8 @@
> >      "url": "https://github.com/pdfme/pdfme/issues"
> >    },
> >    "main": "dist/cjs/src/index.js",
> > -  "module": "dist/esm/src/index.js",
> > +  "module": "dist/cjs/src/index.js",
> >    "types": "dist/types/src/index.d.ts",
> > -  "exports": {
> > -    ".": {
> > -      "import": "./dist/esm/src/index.js",
> > -      "require": "./dist/cjs/src/index.js"
> > -    }
> > -  },
> >    "engines": {
> >      "node": ">=16"
> >    },
> > ```
> 
> This also worked for me.

This is alas not working anymore

```
✘ [ERROR] Could not resolve "@pdfme/schemas"
    ../../node_modules/.pnpm/@pdfme+generator@4.0.1_patch_hash=6mgkat63vj7b5gz6mjxsu3tnwi_@pdfme+common@4.0.1/node_modules/@pdfme/generator/dist/esm/src/generate.js:3:65:
      3 │ ...TemplateForTable, getDynamicHeightForTable } from '@pdfme/schemas';
        ╵                                                      ~~~~~~~~~~~~~~~~
  You can mark the path "@pdfme/schemas" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
✘ [ERROR] Could not resolve "@pdfme/schemas"
    ../../node_modules/.pnpm/@pdfme+generator@4.0.1_patch_hash=6mgkat63vj7b5gz6mjxsu3tnwi_@pdfme+common@4.0.1/node_modules/@pdfme/generator/dist/esm/src/helper.js:3:31:
      3 │ import { builtInPlugins } from '@pdfme/schemas';
        ╵                                ~~~~~~~~~~~~~~~~
  You can mark the path "@pdfme/schemas" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
```

**Comment by FredVicente** [comment]
> > > I use typescript and my tsconfig is like that
> > > ```
> > > {
> > >   "compilerOptions": {
> > >     /* Visit https://aka.ms/tsconfig to read more about this file */
> > >     "target": "ES2020",
> > >     "lib": ["ES2020"],
> > >     "module": "ES2020",
> > >     "outDir": "./dist",
> > >     "rootDir": "./src",
> > >     "removeComments": true,
> > >     "esModuleInterop": true,
> > >     "forceConsistentCasingInFileNames": true,
> > >     "moduleResolution": "node",
> > >     "strict": true,
> > >     "skipLibCheck": true,
> > >   },
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And to successfully use the lib i needed to apply this patch, to both common and generator
> > > ```
> > > diff --git a/node_modules/@pdfme/common/package.json b/node_modules/@pdfme/common/package.json
> > > index 2725927..df0fc38 100644
> > > --- a/node_modules/@pdfme/common/package.json
> > > +++ b/node_modules/@pdfme/common/package.json
> > > @@ -22,14 +22,8 @@
> > >      "url": "https://github.com/pdfme/pdfme/issues"
> > >    },
> > >    "main": "dist/cjs/src/index.js",
> > > -  "module": "dist/esm/src/index.js",
> > > +  "module": "dist/cjs/src/index.js",
> > >    "types": "dist/types/src/index.d.ts",
> > > -  "exports": {
> > > -    ".": {
> > > -      "import": "./dist/esm/src/index.js",
> > > -      "require": "./dist/cjs/src/index.js"
> > > -    }
> > > -  },
> > >    "engines": {
> > >      "node": ">=16"
> > >    },
> > > ```
> > 
> > 
> > This also worked for me.
> 
> This is alas not working anymore
> 
> ```
> ✘ [ERROR] Could not resolve "@pdfme/schemas"
>     ../../node_modules/.pnpm/@pdfme+generator@4.0.1_patch_hash=6mgkat63vj7b5gz6mjxsu3tnwi_@pdfme+common@4.0.1/node_modules/@pdfme/generator/dist/esm/src/generate.js:3:65:
>       3 │ ...TemplateForTable, getDynamicHeightForTable } from '@pdfme/schemas';
>         ╵                                                      ~~~~~~~~~~~~~~~~
>   You can mark the path "@pdfme/schemas" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
> ✘ [ERROR] Could not resolve "@pdfme/schemas"
>     ../../node_modules/.pnpm/@pdfme+generator@4.0.1_patch_hash=6mgkat63vj7b5gz6mjxsu3tnwi_@pdfme+common@4.0.1/node_modules/@pdfme/generator/dist/esm/src/helper.js:3:31:
>       3 │ import { builtInPlugins } from '@pdfme/schemas';
>         ╵                                ~~~~~~~~~~~~~~~~
>   You can mark the path "@pdfme/schemas" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
> ```

I got the same error. Just did the changes that @AugustoResende said in "schemas" too and it worked for me

**Comment by bitsmakerde** [comment]
I have the same with Astro. How it can be open other one year. 

The code looks of it supports esm, but it doesn't.

**Comment by peteward** [comment]
It's open because no one has volunteered a working pull request and the authors of this project do not use Svelte.
If you are confident in a fix and want to help others then please do open a working PR.

**Comment by bitsmakerde** [comment]
I can't find a working solution I use Astro. So the bug isn't connect to one framework. 

If I have a solution I will do a RP, but all solution here not working for me. 

The schema don't like then you do the change with delete module and use cjs. 

Is this tool working in some combinations with vite and esbuild?

**Comment by formalism** [comment]
I'm also in a trouble using the library in my Remix project (which uses Vite). I've tried the method @FredVicente commented above, but unsuccessful. I used the latest version of the library from npm registry as of today. Lastly I tried the above PR locally and found out that the problem has gone away.

**Comment by hand-dot** [comment]
[PR](https://github.com/pdfme/pdfme/pull/617) from @bitsmakerde is merged and Released as 5.1.7

Please check latest pdfme. If we can close this PR, let me know.
